Closed Safaricom Mpesa Daraja (Independent Publisher)

muchirimaina opened the pull request:
---
### When submitting a connector, please make sure that you follow the requirements below, otherwise your PR might be rejected. We want to make you have a well-built connector, a smooth certification experience, and your users are happy :) 

If this is your first time submitting to GitHub and you need some help, please sign up for this [session](https://forms.office.com/pages/responsepage.aspx?id=KtIy2vgLW0SOgZbwvQuRaXDXyCl9DkBHq4A2OG7uLpdUMTFJWFFGVUxBNUFZQjZWRUdaOE5BMFkwNS4u). 

- [x] I attest that the connector doesn't exist on the Power Platform today. I've verified by checking the pull requests in GitHub and by searching for the connector on the platform or in the documentation. 
- [x] I attest that the connector works and I verified by deploying and testing all the operations. 
- [x] I attest that I have added detailed descriptions for all operations and parameters in the swagger file.
- [x] I attest that I have added response schemas to my actions, unless the response schema is dynamic. 
- [x] I validated the swagger file, `apiDefinition.swagger.json`, by running `paconn validate` command.
- [x] If this is a certified connector, I confirm that `apiProperties.json` has a valid brand color and doesn't use an invalid brand color, `#007ee5` or `#ffffff`. If this is an independent publisher connector, I confirm that I am not submitting a connector icon.

If you are an Independent Publisher, you must also attest to the following to ensure a smooth publishing process:
- [x] I have named this PR after the pattern of "Connector Name (Independent Publisher)" ex: HubSpot Marketing (Independent Publisher)
- [x] Within this PR markdown file, I have pasted screenshots that show: 3 unique operations (actions/triggers) working within a Flow. This can be in one flow or part of multiple flows. For each one of those flows, I have pasted in screenshots of the Flow succeeding. 
- [x] Within this PR markdown file, I have pasted in a screenshot from the Test operations section within the Custom Connector UI.
- [x] If the connector uses OAuth, I have provided detailed steps on how to create an app in the readme.md.

94farhan committed:
Partner Center connector - new fields and deal registration actions (#2130)

* Update apiDefinition.swagger.json

* Latest changes

* fix format warning

* X-Caller_Id should be optional

* Update apiDefinition.swagger.json

Review changes -
->Added reg status and sub status fields.
->Update the fields descriptions as per api's details available on Microsoft public site.
->Remove submission related fields.
->Mark registration status field as read only.
->Remove offerId from root.
->Use Enum values in Create deal registration model.
->Create a separate model for update deal registration.
Create a separate action for update deal registration.

* Update apiDefinition.swagger.json

* Update apiDefinition.swagger.json

Add deprecate text in Update a referral by Id action

* Update apiDefinition.swagger.json

Connector status keep as Preview and UpdateReferralById deprecated keep as false but deprecated word in description.

* Update apiDefinition.swagger.json

update deal registration success message.

* Update apiDefinition.swagger.json

added title for registration object

saferoombiz committed:
Bitskout Connector (#2140)

* Bitskout Connector

* Requested changes

thomasio101 committed:
EC-14395 Make it so fields aren't required anymore for updates (#2150)

* Added orderby parameter

* Added orderby parameter

* feat: add x-ms-pageable extension

* feat: add updatenextlink policy instance

* refactor: whitespace refactoring

* Update apiProperties.json

* fix(EC-14395): Integrate the patch schema endpoint

* fix(EC-14395): Add specs for the schema endpoint's responses

* fix(EC-14395): Add required fields to the schema

Co-authored-by: Johannes Heesterman <johannes@elfsquad.io>
Co-authored-by: johannesheesterman <heestermanj@gmail.com>
Co-authored-by: Stan van Rooy <stanvanrooy6@gmail.com>

mamurshe committed:
Merge branch 'dev'

muchirimaina committed:
[Daraja] first commit

johnmuchiriM commented:
The test operations were done and all passed.
![daraja](https://user-images.githubusercontent.com/60730009/229096211-2ef9b110-df9c-4dca-b713-12534bbd0bd8.jpeg)

johnmuchiriM commented:
Validation of the connector passed successfully using the paconn tool
![daraja-validation](https://user-images.githubusercontent.com/60730009/229096524-776a6f5b-0b04-44fa-a2dc-d0674a0c66d0.jpeg)

johnmuchiriM commented:
The stk push on the phone when testing the operations.
![Uploading darajaStkpush.jpeg…]()

johnmuchiriM commented:
A simple flow of one operation test. "Lipa na Mpesa" 
![darajaFlow](https://user-images.githubusercontent.com/60730009/229097616-40f072ab-3ad8-4a67-91d8-ac28a0700224.jpeg)

johnmuchiriM commented:
Hi @Amjed-Ayoub,

I would like to request a review of the submitted connector. Also, my colleague and I are working on this connector as a team of two developers.

Could you please advise on how to add my colleague, whose email is [bkaraba14@gmail.com](mailto:bkaraba14@gmail.com), to the communication thread?

Lastly, I have a question regarding creating a pull request. After forking and cloning the repository, and creating a new branch, I noticed that this particular pull request includes some other commits, which I will attach in an image. I'm not sure how these commits ended up in my new pull request. Could you help me figure this out?

Thank you very much.
![Screenshot from 2023-04-05 00-38-57](https://user-images.githubusercontent.com/60730009/229928791-36e2a366-c689-4cc8-801b-4cf408fbb53e.png)

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,620 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "daraja",
```
@johnmuchiriM is it possible to capitalize the first letter of the title?

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,620 @@
+      "name": "John Muchiri, Bernard Karaba",
+      "url": "https://johnmuchirim.github.io/",
+      "email": "jeanfrancaiseharicot@gmail.com",
+      "email": "bkaraba14@gmail.com"
```
@johnmuchiriM please have both emails in one property. please see below.
 "email": "jeanfrancaiseharicot@gmail.com, bkaraba14@gmail.com"

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,620 @@
+    "application/json"
+  ],
+  "paths": {
+    "/oauth/v1/generate": {},
```
@johnmuchiriM can you please explain the empty operations? is it possible to remove it?
Please fix all empty paths in the swagger file.

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,620 @@
+            }
+          }
+        },
+        "summary": "lipa na mpesa",
```
@johnmuchiriM all information within the swagger file should be in English language.

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,620 @@
+        "responses": {
+          "200": {
+            "description": "default",
+            "schema": {}
```
@johnmuchiriM empty schemas are not allowed. please define a proper schema. Please see [here](https://learn.microsoft.com/en-us/connectors/custom-connectors/coding-standards#responses). 
Please fix all empty schemas in the swagger file.

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,620 @@
+   "x-ms-connector-metadata": [
+    {
+      "propertyName": "Website",
+      "propertyValue": "https://www.developer.safaricom.co.ke/"
```
@johnmuchiriM this should be a valid URL. 
![image](https://user-images.githubusercontent.com/97551789/230199095-a3e17057-aa2f-4056-9490-ec2636da7802.png)

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,620 @@
+    },
+    {
+      "propertyName": "Privacy policy",
+      "propertyValue": "https://www.developer.safaricom.co.ke"
```
@johnmuchiriM this should link to the privacy policy page please.

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiProperties.json:
```diff
@@ -0,0 +1,48 @@
+        }
+      }
+    ],
+    "publisher": "Bernard Karaba"
```
@johnmuchiriM you can add both publisher here as well. Please see below.
"publisher": "Publisher1, Publisher2"

Please add the stackOwner property. Please see our template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json#:~:text=%22stackOwner%22%3A%20%22Name%20of%20company%20or%20organization%20that%20connector%20is%20connecting%20to.%22). 
"stackOwner": "Name of company or organization that connector is connecting to."

muchirimaina committed:
[Daraja] updated readme file

Amjed-Ayoub commented:
> Hi @Amjed-Ayoub,
> 
> I would like to request a review of the submitted connector. Also, my colleague and I are working on this connector as a team of two developers.
> 
> Could you please advise on how to add my colleague, whose email is [bkaraba14@gmail.com](mailto:bkaraba14@gmail.com), to the communication thread?
> 
> Lastly, I have a question regarding creating a pull request. After forking and cloning the repository, and creating a new branch, I noticed that this particular pull request includes some other commits, which I will attach in an image. I'm not sure how these commits ended up in my new pull request. Could you help me figure this out?
> 
> Thank you very much. ![Screenshot from 2023-04-05 00-38-57](https://user-images.githubusercontent.com/60730009/229928791-36e2a366-c689-4cc8-801b-4cf408fbb53e.png)

@johnmuchiriM I will include @Bkaraba in all comments in this PR.

Please ignore the previous commits. and make sure to update your branch every time you would like to submit a new PR.

johnmuchiriM commented:
Hi @Amjed-Ayoub,
I have noted all the requested changes and am currently working on them. I will notify you as soon as everything is resolved. I am excited to work on this particular connector and it's a great experience. I am sure my partner is also excited to work on this connector.
Thank you so much.

muchirimaina committed:
[Daraja] resolved requested changes

muchirimaina committed:
[Daraja] made some small change on some grammar error

johnmuchiriM commented:
Hi @Amjed-Ayoub,
Everything from earlier conversation has been resolved.

muchirimaina committed:
[Daraja] removed empty operation

johnmuchiriM commented:
Hi @Amjed-Ayoub,
Removed the empty operation.
Regards,
John.

Amjed-Ayoub commented:
[[certify-connector]]

johnmuchiriM commented:
Hi @Amjed-Ayoub, 
I'm doing well, thank you for asking. My partner is also doing great. These are great news! Thank you very much for your support throughout the whole process! I'm excited to see the next steps.

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiDefinition.swagger.json:
```diff
@@ -0,0 +1,696 @@
+    "contact": {
+      "name": "John Muchiri, Bernard Karaba",
+      "url": "https://johnmuchirim.github.io/",
+      "email": "jeanfrancaiseharicot@gmail.com"
```
@johnmuchiriM and @Bkaraba please notice that we have only one email address here. please confirm?

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiProperties.json:
```diff
@@ -0,0 +1,49 @@
+        "type": "oauthSetting",
+        "oAuthSettings": {
+          "identityProvider": "oauth2",
+          "clientId": "4b048dda-9dc5-412b-b881-1f6881f882d8",
```
@johnmuchiriM and @Bkaraba secret look alike values are not allowed. please replace the client ID with a dummy value. You can use something like "Enter your client ID"

Amjed-Ayoub commented on independent-publisher-connectors/Safaricom Mpesa Daraja/apiProperties.json:
```diff
@@ -0,0 +1,49 @@
+        }
+      }
+    },
+    "iconBrandColor": "#007ee5",
```
@johnmuchiriM and @Bkaraba for all our Independent Publisher we ask the iconBrandColor to be "#da3b01". can you please change it. please see out template [here](https://github.com/microsoft/PowerPlatformConnectors/blob/dev/templates/Independent%20Publisher/apiProperties.json).

muchirimaina committed:
[Daraja] added two emails,added dummy client id,changed iconBrandColor to default

Amjed-Ayoub commented:
Hello @johnmuchiriM and @Bkaraba,
I hope you are doing well.
Congratulations, your pull request is approved. We will move forward with the certification process. We are preparing your connector for production deployment and will queue up for the next deployment schedule. Once your connector onboards the next deployment schedule, it starts to deploy your connector in our production environments that typically takes 3-4 weeks. 
Please let us know if you have any questions.
Thank you very much for working with us.

johnmuchiriM commented:
Hi @Amjed-Ayoub,
These are great news! We're super excited to see the next phase!! 
Thank you so much!

Amjed-Ayoub commented:
[[certify-connector]]

Amjed-Ayoub commented:
Hello @johnmuchiriM ,
I hope you are doing well.
Your connector is in production deployment and queued up for the next deployment schedule. This process typically takes 3-4 weeks. Please note that the documentation for your connector will not be generated until the deployment process is in the final steps.
Please let us know if you have any questions.
Thank you very much for working with us.

muchirimaina commented:
Hi @Amjed-Ayoub 

The news gives off such a positive vibe as we head into the weekend. Thanks for your support as we work together.

Have a great weekend ahead!

muchirimaina commented:
Hi @mamurshe & @Amjed-Ayoub,
I hope you're doing well. I'm just checking in to inquire about the connector deployment status. It's been a while since I heard from the team. Any updates would be greatly appreciated.
Thanks and have a great day!

SaiChander-MS commented:
@johnmuchiriM While we started the deployment process, we see few gaps in your submission like the steps to create AAD application is invalid in readme.md file. Also, the authorizationUrl and host Url in the commit are invalid.
Please address the review comments, so that we can proceed with next steps.

vmanoharas commented:
Hello @johnmuchiriM, 

Please let's know have you addressed the comments?

muchirimaina commented:
Hi @vmanoharas,

I have addressed the comments, but I am facing **a permission error** when I try to push to my branch, which I was using before. I would appreciate an extra set of eyes on this. I have **attached the error**, the **Git log showing previous successful commits**, and my **Git config file image**. For the password, I'm using the personal token from developer settings.

Thanks

![Screenshot from 2023-09-10 21-16-03](https://github.com/microsoft/PowerPlatformConnectors/assets/60730009/5c6b7eee-7c27-481c-8551-d6ab6e2e743f)
![Screenshot from 2023-09-10 21-15-33](https://github.com/microsoft/PowerPlatformConnectors/assets/60730009/c3c01d31-b766-4b1d-9127-ced13dcb10cb)
![Screenshot from 2023-09-10 21-14-33](https://github.com/microsoft/PowerPlatformConnectors/assets/60730009/80e386b8-44c5-4099-b73f-d7cc5186df38)
![Screenshot from 2023-09-10 21-49-43](https://github.com/microsoft/PowerPlatformConnectors/assets/60730009/b4f92af0-3db0-4128-978b-8801c085520e)

vmanoharas commented:
Hello @johnmuchiriM, 

Can you please close this PR and submit new PR?

muchirimaina commented:
Hello @vmanoharas 
That would be great. Let me proceed with that!

muchirimaina commented:
I closed this branch in order to open a new one for submitting the same connector.
